### PR TITLE
samples: bluetooth: BTHome sensor template fix whitespace

### PR DESCRIPTION
**Pull request by osaether**
Whitespace fix.

### CODE REVIEW

**Comment by stephanosio** [comment]
Please rebase to fix the manifest and assigner workflow failures.

File: samples/bluetooth/bthome_sensor_template/src/main.c [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include <zephyr/bluetooth/uuid.h>
 
 #define SERVICE_DATA_LEN        9
-#define SERVICE_UUID            0xfcd2		/* BTHome service UUID */
+#define SERVICE_UUID            0xfcd2      /* BTHome service UUID */
 #define IDX_TEMPL               4           /* Index of lo byte of temp in service data*/
 #define IDX_TEMPH               5           /* Index of hi byte of temp in service data*/
```
Comment (jori-nordic) [inline samples/bluetooth/bthome_sensor_template/src/main.c:14]: can you also fix the ADV_PARAM whitespace/alignment?

**Comment by jori-nordic** [comment]
image

@osaether I think you need to tab until you can't, and then spaces to align the remaining.

**Review by stephanosio — DISMISSED** [review]

**Comment by jori-nordic** [comment]
@osaether I pushed what I wanted if you don't mind. You just have to configure your editor to use the zephyr-provided .editorconfig, and it will do the right thing (start indenting w/ tabs instead of spaces).

**Review by jori-nordic — DISMISSED** [review]

**Comment by jhedberg** [comment]
> Zephyr does not accept spaces at the start of a line so I will leave the (already approved) indent as it was. So now this PR is back to fix the single line.

While I don't think it's enforced anywhere, 0-N tabs in the beginning of a line is the *only* place where tabs should ever be found. Using spaces everywhere else tends to ensure that the alignment looks right regardless of if you're viewing the code in a web browser (on GitHub) or in your editor. Right now the PR adds tabs before `\` on a couple of lines, which violates this rule.

**Comment by jori-nordic** [comment]
@jhedberg where's that rule?

**Comment by jhedberg** [comment]
> @jhedberg where's that rule?

Maybe I should have been clearer: it's neither enforced nor documented as a rule, as far as I know. However I'm recommending to follow it, since the practical consequences are as I outlined (we get the least messy result from having both tabs and spaces in this project).

**Comment by jhedberg** [comment]
> edit: have to add that right now the code looks "right" on github & in both emacs (that loads editorconfig) & vim (that doesn't).

It really depends on luck, i.e. at what column of the line do you start adding tabs again (after having some non-tab characters).

**Comment by jori-nordic** [comment]
> I'm recommending to follow it

Not to be harsh, but that's just your personal preference at this point. If it's not enforceable/ed, we should at the very least document it somewhere as a project preference. 
However I do agree with the rule you're proposing, I pushed an update removing those tabs.

**Review by jhedberg — APPROVED** [review]

**Comment by carlescufi** [comment]
> > I'm recommending to follow it
> 
> Not to be harsh, but that's just your personal preference at this point. If it's not enforceable/ed, we should at the very least document it somewhere as a project preference. However I do agree with the rule you're proposing, I pushed an update removing those tabs.

It's a bit more than a personal preference though: It is a rule that we've tacitly agreed to in the Bluetooth subsystem for years now, with most (if not all) contributors adhering to it.
Sometimes unwritten rules can be rules as well, and I do think this is one of these cases.

**Review by jori-nordic — APPROVED** [review]

**Pull request merged**